pkg/scheduler/scheduler.go

```diff
@@ -203,6 +204,21 @@ func (e *entry) assignFlavors(log logr.Logger, cap *cache.ClusterQueue) bool {
 // assuming it in the cache.
 func (s *Scheduler) admit(ctx context.Context, e *entry) error {
 	log := ctrl.LoggerFrom(ctx)
+
+	// Check if the workload exists, if not skip assume it in cache.
+	found := &kueue.QueuedWorkload{}
```

Comment:
We can avoid `assume + update + forget` if we can find it doesn't exist by pre-checking. It may be like checking if the pod exists at the start of the scheduling cycle in kube-scheduler.

Reply:
ok, but then this is not a "bug" more of an optimization (which I am fine with), correct?

Reply:
yes. I remove the label of `bug`.

Reply:
Ah true, I missed that it would fail later when I opened the issue #92.

I think it's fine to leave the code as is, for simplicity.